### ISSUE TITLE
Update cli argument parser to warn about ignored/invalid arguments instead of exit

### ISSUE THREAD

**Issue by michaeljota** [angular/angular-cli#13363]
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [x] feature request
```

### Command (mark with an `x`)
```
- [ ] new
- [x] build
- [x] serve
- [x] test
- [x] e2e
- [ ] generate
- [ ] add
- [ ] update
- [x] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Versions

```
node -v; npm -v; ng v
v10.14.1
6.4.1

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/


Angular CLI: 7.1.0
Node: 10.14.1
OS: win32 x64
Angular:
...

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.11.0
@angular-devkit/core         7.1.0
@angular-devkit/schematics   7.1.0
@schematics/angular          7.1.0
@schematics/update           0.11.0
rxjs                         6.3.3
typescript                   3.1.6
```

### Repro steps

Run any cli command with an argument that command should not be able to understand, even when that command builder was changed.

---

Using [@angular-builders/jest](https://github.com/meltedspark/angular-builders/tree/master/packages/jest):

```
ng test my-app my-component.ts
```

### The log given by the failure

```
Unknown option: my-component.ts
```

### Desired functionality

I would like this to be parsed by the builder, as the builder would/should understand that argument.

I understand why the change was introduced, as if you pass an invalid argument to a builder command, or a tester command it's most likely an error, but I really think you should take in consideration the number of issues opened by this stricter parser. (#12305, #13078, Sure there are several others). 

I am NOT asking to support passing this arguments to the default builders, as #11408 or #12299. This builders should ignore those arguments, and give a warning.

**Comment by alan-agius4** [comment]
Hi there,

The actual parser does support positional arguments like `ng test my-app my-component.ts` If they are declared in `schema.json`.


Example;

```JSON
"name": {
	"type": "string",
	"$default": {
	"$source": "argv",
	   "index": 0
	}
},
"file": {
	"type": "string",
	"$default": {
	"$source": "argv",
	   "index": 1
	}
}
```

**Comment by michaeljota** [comment]
@alan-agius4 How can this be doable without knowing the exactly position of the argument? I'm using [vscode-jest](https://github.com/jest-community/vscode-jest) to run the test, it was tricky to setup, but after some research I did it, and it was working everything before the update, but after that I had to update some scripts and lastly I lose the ability to `debug` failing test suits. 

This is actually one the best features the plugin have, and I really want to be able to do it again.
